diff --git a/riscV/Stacklayout.v b/riscV/Stacklayout.v
index 25f02aab..059d5fc1 100644
--- a/riscV/Stacklayout.v
+++ b/riscV/Stacklayout.v
@@ -59,7 +59,7 @@ Lemma frame_env_separated:
        ** range sp (fe_ofs_callee_save fe) (size_callee_save_area b (fe_ofs_callee_save fe))
        ** P.
 Proof.
-Local Opaque Z.add Z.mul sepconj range.
+Local Opaque Z.add Z.mul sepconj range'.
   intros; simpl.
   set (w := if Archi.ptr64 then 8 else 4).
   set (olink := align (4 * b.(bound_outgoing)) w).
